### ISSUE TITLE
Numpy + external data support

### ISSUE THREAD

**Issue by UnamedRus** [mymarilyn/clickhouse-driver#267]
**Describe the bug**
Looks like it's not possible to send external_data with enabled numpy support.

**To Reproduce**
```
from datetime import date, datetime
from ipaddress import ip_address
from time import time
from uuid import UUID
import numpy as np
import pandas as pd

from clickhouse_driver import Client


def run_query():
    client = Client(host = "localhost", port = 9000, database ="default", user  = "default", settings={'use_numpy': True})
    df = pd.DataFrame({
        'x': [100,500],
        'y': [1, 2],
    }, dtype=object)

    tables = [{
        'name': 'ext',
        'structure': [('x', 'Int32'), ('y', 'Int32')],
        'data': [
            {'x': 100, 'y': 2},
            {'x': 500, 'y': 3},
        ]
    }]

    tables2 = [{
        'name': 'ext',
        'structure': [('x', 'Int32'), ('y', 'Int32')],
        'data': df
    }]

    res = client.execute(
        'SELECT sum(x) FROM ext', external_tables=tables2, columnar=False
    )
    print(res)

if __name__ == "__main__":
    run_query();
```

```
Traceback (most recent call last):
  File "./testasync.py", line 39, in <module>
    run_query();
  File "./testasync.py", line 33, in run_query
    res = client.execute(
  File "/home/xxx/.local/lib/python3.8/site-packages/clickhouse_driver/client.py", line 286, in execute
    rv = self.process_ordinary_query(
  File "/home/xxx/.local/lib/python3.8/site-packages/clickhouse_driver/client.py", line 477, in process_ordinary_query
    self.connection.send_external_tables(external_tables,
  File "/home/xxx/.local/lib/python3.8/site-packages/clickhouse_driver/connection.py", line 616, in send_external_tables
    block = RowOrientedBlock(table['structure'], table['data'],
  File "/home/xxx/.local/lib/python3.8/site-packages/clickhouse_driver/block.py", line 39, in __init__
    self.data = self.normalize(data or [])
  File "/home/xxx/.local/lib/python3.8/site-packages/pandas/core/generic.py", line 1478, in __nonzero__
    raise ValueError(
ValueError: The truth value of a DataFrame is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().

```

If we will switch external_tables to tables:

```
Traceback (most recent call last):
  File "./testasync.py", line 39, in <module>
    run_query();
  File "./testasync.py", line 33, in run_query
    res = client.execute(
  File "/home/xxx/.local/lib/python3.8/site-packages/clickhouse_driver/client.py", line 286, in execute
    rv = self.process_ordinary_query(
  File "/home/xxx/.local/lib/python3.8/site-packages/clickhouse_driver/client.py", line 477, in process_ordinary_query
    self.connection.send_external_tables(external_tables,
  File "/home/xxx/.local/lib/python3.8/site-packages/clickhouse_driver/connection.py", line 618, in send_external_tables
    self.send_data(block, table_name=table['name'])
  File "/home/xxx/.local/lib/python3.8/site-packages/clickhouse_driver/connection.py", line 567, in send_data
    self.block_out.write(block)
  File "/home/xxx/.local/lib/python3.8/site-packages/clickhouse_driver/streams/native.py", line 38, in write
    write_column(self.context, col_name, col_type, items,
  File "/home/xxx/.local/lib/python3.8/site-packages/clickhouse_driver/columns/service.py", line 113, in write_column
    column.write_data(items, buf)
  File "/home/xxx/.local/lib/python3.8/site-packages/clickhouse_driver/columns/base.py", line 77, in write_data
    self._write_data(items, buf)
  File "/home/xxx/.local/lib/python3.8/site-packages/clickhouse_driver/columns/base.py", line 81, in _write_data
    self.write_items(prepared, buf)
  File "/home/xxx/.local/lib/python3.8/site-packages/clickhouse_driver/columns/numpy/base.py", line 17, in write_items
    buf.write(items.astype(self.dtype.newbyteorder('<')).tobytes())
AttributeError: 'list' object has no attribute 'astype'
```

**Expected behavior**
Working example or bug fix :) 

**Versions**

clickhouse-driver-0.2.2
ClickHouse 21.12
Python 3.8.5 (default, Jul 28 2020, 12:59:40)

**Comment by xzkostyan** [comment]
Fixed in 0.2.3.

**Comment by UnamedRus** [comment]
It seems, not example with `tables` doesn't work with 

```
 settings={'use_numpy': True}
 'SELECT sum(x) FROM ext', external_tables=tables, columnar=False

Traceback (most recent call last):
  File "./testasync.py", line 39, in <module>
    run_query();
  File "./testasync.py", line 33, in run_query
    res = client.execute(
  File "/home/unamed/.local/lib/python3.8/site-packages/clickhouse_driver/client.py", line 296, in execute
    rv = self.process_ordinary_query(
  File "/home/unamed/.local/lib/python3.8/site-packages/clickhouse_driver/client.py", line 480, in process_ordinary_query
    self.connection.send_external_tables(external_tables,
  File "/home/unamed/.local/lib/python3.8/site-packages/clickhouse_driver/connection.py", line 632, in send_external_tables
    data = [data[column].values for column in columns]
  File "/home/unamed/.local/lib/python3.8/site-packages/clickhouse_driver/connection.py", line 632, in <listcomp>
    data = [data[column].values for column in columns]
TypeError: list indices must be integers or slices, not str
```
